### ISSUE TITLE
Metadata Attribute validation: DelegationRole attributes

### ISSUE THREAD

**Issue by MVrachev** [theupdateframework/tuf#1461]
The targets class is implemented as a combination of two classes - `TargetFile` and `Delegations` plus the common signed attributes.

All Signed attributes are validated in various prs:
- [https://github.com/theupdateframework/tuf/pull/1430](https://github.com/theupdateframework/tuf/pull/1430) 
- [https://github.com/theupdateframework/tuf/issues/1418](https://github.com/theupdateframework/tuf/issues/1418)
- [https://github.com/theupdateframework/tuf/issues/1420](https://github.com/theupdateframework/tuf/issues/1420)

We already have validation for `TargetFile` attributes added thanks to @sechkova in [https://github.com/theupdateframework/tuf/pull/1451](https://github.com/theupdateframework/tuf/pull/1451).

For `Delegations` class. we have `keys` which are mappings of a string to a `Key` instance and we have already added validation for the `Key` instances here: [https://github.com/theupdateframework/tuf/pull/1449](https://github.com/theupdateframework/tuf/pull/1449).

Actually, the only piece class left for validation is the one responsible for `roles` in `Delegations` or `DelegatedRole`.

From the `DelegatedRole` attributes we are already validating `threshold` and `keyids` inside the parent class `Role` which we added here [https://github.com/theupdateframework/tuf/pull/1450](https://github.com/theupdateframework/tuf/pull/1450) and validation for `keyid` is added with the Key class validation  [https://github.com/theupdateframework/tuf/pull/1449](https://github.com/theupdateframework/tuf/pull/1449).

**So, we have to decide how we are going to validate `name`, `paths` and `path_hash_prefixes` attributes from `DelegatedRole`.**

**Comment by MVrachev** [comment]
We shouldn't forget to add tests for invalid attributes the way we did for the rest of the attributes here: #1460.

**Comment by MVrachev** [comment]
DelegatedRole name is discussed separately in #1558.

**Comment by MVrachev** [comment]
We had a good discussion with @jku about `paths` and `path_hash_prefixes`.
The conclusions from our conversation are:

1. Even though adding a `""` in `path_hash_prefix` in a delegated role may be a bit of a hack it feels quite straightforward.
This seems to be  like currently the only way to delegate absolutely everything.
2. I tried adding a `""` in `paths` and it didn't work as [DelegatedRole._is_target_in_pathpattern](https://github.com/theupdateframework/python-tuf/blob/1b5df4ccfec6306e841b0573aa4a0e26ce0846d2/tuf/api/metadata.py#L1141) doesn't allow using `""` as a path.

So, to conclude we decided we want to validate only `paths` and `path_hash_prefixes` types and that's the only TODO item for this issue.
